src/Database/Type/ExpressionTypeCasterTrait.php

```diff
@@ -49,7 +49,7 @@ protected function _castToExpression($value, ?string $type = null)
         $multi = $type !== $baseType;
 
         if ($multi) {
-            /** @psalm-suppress InvalidArgument */
+            /** @var \Cake\Database\Type\ExpressionTypeInterface $converter */
```

Comment:
IDEs might not like this too without the variable assignment, so could use `@psalm-var` instead.

Reply:
I mean, they wouldn't throw an error though, right?

Reply:
Changed to psalm-var for now since it's a special case.